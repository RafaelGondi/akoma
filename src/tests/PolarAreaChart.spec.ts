import { describe, test, expect } from 'vitest';
import PolarAreaChart from '../components/PolarAreaChart.vue';
import { shallowMount } from '@vue/test-utils';

const mockedData = [
	{
		name: 'Ecocardiograma',
		datasets: [
			{
				label: 'Exames realizados',
				data: [120, 220, 180, 320, 150],
			},
		]
	},
];

describe('PolarAreaChart', () => {
	test('renders correctly', async () => {
		const wrapper = shallowMount(PolarAreaChart, {
			props: {
				labels: ['Janeiro','Fevereiro','Mar√ßo', 'Abril', 'Maio'],
				variant: 'green',
				data: mockedData,
				isVisiblePointNames: true,
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
