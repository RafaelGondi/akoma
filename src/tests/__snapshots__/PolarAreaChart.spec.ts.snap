// Vitest Snapshot v1

exports[`PolarAreaChart > renders correctly 1`] = `"<span data-v-d3067411=\\"\\"><div class=\\"responsive-container\\" data-v-d3067411=\\"\\"><polar-area-stub data=\\"[object Object]\\" options=\\"[object Object]\\" plugins=\\"\\" datasetidkey=\\"label\\" data-v-d3067411=\\"\\"></polar-area-stub></div></span>"`;
